dev/breeze/src/airflow_breeze/utils/selective_checks.py

```diff
@@ -1767,3 +1768,125 @@ def parse_dep(dep_str: str) -> tuple[str, str | None]:
                             )
 
         return violations
+
+    @cached_property
+    def common_compat_changed_without_next_version(self) -> bool:
+        """
+        Check if common.compat provider changed and other providers changed don't have '# use next version'
+        comment for their common-compat dependency.
+        """
+        if self._github_event != GithubEvents.PULL_REQUEST:
+            return False
+
+        # First, check if common.compat provider was changed
+        common_compat_changed = False
+        for changed_file in self._files:
+            if changed_file.startswith("providers/common/compat/"):
+                common_compat_changed = True
+                break
+
+        if not common_compat_changed:
+            return False
```

Comment:
Oh and with this following even shorter:
```suggestion
        if not any(f.startswith("providers/common/compat/") for f in self._files):
            return False
```